File: server/Models/Recipe.cs
namespace all_spice.Models;

public class Recipe : RepoItem<int>
{
  public string Title { get; set; }
  public string? Instructions { get; set; }
  // public List<Ingredient> Ingredients { get; set; }
  public string Img { get; set; }
  public string Category { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
}

File: client/src/models/Recipe.js
import { AppState } from "@/AppState.js"
import { Profile } from "./Account.js"

export class Recipe {
  constructor(data) {
    this.id = data.id
    this.createdAt = data.createdAt
    this.updatedAt = data.updatedAt
    this.title = data.title
    this.instructions = data.instructions
    this.img = data.img
    this.category = data.category
    this.creatorId = data.creatorId
    this.creator = new Profile(data.creator)
  }
  get isFavorite() {
    const recipes = AppState.favoriteRecipes
    if (recipes.find(recipe => recipe.id == this.id)) {
      return true
    }
    return false
  }
}

export class FavoriteRecipe extends Recipe {

  constructor(data) {
    super(data);
    this.favoriteId = data.favoriteId
  }
}